gcli/src/cmd/reply.rs

```diff
@@ -52,7 +52,9 @@ impl Reply {
         let signer = app.signer().await?;
         let reply_to_id = self.reply_to_id.to_hash()?;
 
-        let gas_limit = if self.gas_limit == 0 {
+        let gas_limit = if let Some(gas_limit) = self.gas_limit {
+            gas_limit
+        } else {
```

Comment:
I'd use `unwrap_or_else`.
```suggestion
        let gas_limit  = self.gas_limit.unwrap_or_else(|| {
```

Reply:
calculating gas requires `async` methods, so it is hard to be placed in the `else` closure even it is cleaner

Reply:
Ah, agree, missed it.